seacrowd/sea_datasets/cosem/cosem.py

```diff
@@ -0,0 +1,220 @@
+    abstract = {Colloquial Singapore English (CSE, commonly known as Singlish) is a linguistic variety used in Singapore, a Southeast
+    Asian nation home to three major ethnic groups: the Chinese (74.35% of the citizen and permanent resident population), the
+    Malays (13.43%), and the Indians (9%) (Singapore Department of Statistics, 2019). It is one of the best known post-colonial
+    varieties of English and has been documented since the emergence of the field of world Englishes (e.g., Greenbaum, 1988; Richards
+    & Tay, 1977). Linguistically, the grammar and lexicon of CSE are systematically imported from other non-English languages used
+    in the island nation (Leimgruber, 2011). From a creolist perspective, it can be viewed as an English-lexifier creole that
+    contains influences from Sinitic languages such as Hokkien, Cantonese and Mandarin, as well as Malay, Tamil and other varieties
+    in the Singapore language ecology (McWhorter, 2007; Platt, 1975). Several distinct features across various levels of language
+    have been investigated in CSE, including phonetics (Starr & Balasubramaniam, 2019), morphosyntax (Bao, 2010; Bao & Wee, 1999
+    semantics (Hiramoto & Sato, 2012), and pragmatics (Hiramoto, 2012; Leimgruber, 2016; Lim, 2007).},
+    language = {en},
+    urldate = {2022-08-08},
+    journal = {English Today},
+    author = {Gonzales, Wilkinson Daniel Wong and Hiramoto, Mie and Leimgruber, Jakob R. E. and Lim, Jun Jie},
+    month = jun,
+    year = {2022},
+    pages = {1--14},
+    file = {Gonzales et al. - 2022 - Is it in Colloquial Singapore English What.pdf:/Users/wdwg/Zotero/storage/59E8FXJT/Gonzales et al. - 2022 - Is it in Colloquial Singapore English What.pdf:application/pdf},
+}
+"""
+
+_DATASETNAME = "cosem"
+
+_DESCRIPTION = """\
+The CoSEM dataset consists of over 900,000 lines of online messages from the messaging platform WhatsApp collected from personal chat
+logs of students enrolled in an advanced sociolinguistics class from the National University of Singapore. Messages collected were
+from 2016 to 2019. The dataset is in .txt format, where each line of utterance is tagged with a unique identifier that includes its
+metadata such as line number, year message was sent, and age and nationality of sender.
+"""
+
+_HOMEPAGE = "https://github.com/wdwgonzales/CoSEM/blob/main/Corpus/COSEM_v4_publicrelease_SEP172023.zip"
+
+_LANGUAGES = ["eng"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+_LICENSE = Licenses.CC0_1_0.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://github.com/wdwgonzales/CoSEM/raw/main/Corpus/COSEM_v4_publicrelease_SEP172023.zip"}
+
+_SUPPORTED_TASKS = [Tasks.SELF_SUPERVISED_PRETRAINING]
+_SUPPORTED_SCHEMA_STRINGS = [f"seacrowd_{str(TASK_TO_SCHEMA[task]).lower()}" for task in _SUPPORTED_TASKS]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class MyWsl2023(datasets.GeneratorBasedBuilder):
```

Comment:
Maybe use CoSEM instead?

```suggestion
class CoSEMDataset(datasets.GeneratorBasedBuilder):
```